### PR TITLE
Add JSON function docs

### PR DESCRIPTION
**Pull request by prrao87**
@acquamarin and @SterlingT3485 please update the JSON docs at the end of this file.

### CODE REVIEW

File: src/content/docs/extensions/json.mdx [hunk 1]
```diff
@@ -548,34 +548,44 @@ Output:
...
+```
+
 ### `json_extract`
 
 Signatures: `JSON, STRING -> JSON`, `JSON, INTEGER -> JSON`
 
 `json_extract(json, path)`
 
-Extracts `JSON` from `json` at the given `path`.  `STRING` `path` is delimited by `'/'`, while `INTEGER` path is only used to index `JSON arrays`. If `path` is a `LIST`, the result will be a `LIST` of `JSON`.
+Extracts `JSON` from `json` at the given `path`.  `STRING` `path` is delimited by `'/'`, while integer path is only used to index JSON arrays. If `path` is a `LIST`, the result will be a `LIST` of JSON.
```
Comment (prrao87) [inline src/content/docs/extensions/json.mdx:588]: @acquamarin this part isn't clear - what does "integer path" even mean? Why would the user provide a path which is an integer?

File: src/content/docs/extensions/json.mdx [hunk 2]
```diff
@@ -548,34 +548,44 @@ Output:
 ```
 
 ```cypher
-RETURN to_json({duck : 42});
+RETURN to_json({kuzu : 42});
```
Comment (prrao87) [inline src/content/docs/extensions/json.mdx:551]: Could we give a more realistic example using one of the example JSON fields from above? Say Person name and age.

File: src/content/docs/extensions/json.mdx [hunk 3]
```diff
@@ -548,34 +548,44 @@ Output:
...
 ```
 Output:
 ```json
-{"duck":42}
+{"kuzu":42}
 ```
 
 ```cypher
-RETURN to_json(map(['duck'],[42]));
+RETURN to_json(map(['kuzu'],[42]));
```
Comment (prrao87) [inline src/content/docs/extensions/json.mdx:559]: Same comment - it's better to use a more realistic example from above. Also, why are both the inputs to the `map` function lists?

File: src/content/docs/extensions/json.mdx [hunk 4]
```diff
@@ -527,7 +527,7 @@ Signature: `ANY -> JSON`
 
 `to_json(any)`
 
-Converts `any` Kuzu value to a `JSON` document
+Converts `any` Kuzu value to `JSON`.
```
Comment (prrao87) [inline src/content/docs/extensions/json.mdx:530]:
```suggestion
Converts `any` Kùzu value to `JSON`.
```

File: src/content/docs/extensions/json.mdx (hunk 1 above)
Comment (SterlingT3485) [reply src/content/docs/extensions/json.mdx:641]: the integer path means the index, let me try rephrase that

File: src/content/docs/extensions/json.mdx (hunk 2 above)
Comment (SterlingT3485) [reply src/content/docs/extensions/json.mdx]: changed to real examples

File: src/content/docs/extensions/json.mdx (hunk 1 above)
Comment (SterlingT3485) [reply src/content/docs/extensions/json.mdx]: re paraphrase this a little

File: src/content/docs/extensions/json.mdx [hunk 5]
```diff
@@ -515,4 +515,288 @@ and `"to"` in the file, whose values point to the primary key values of the unde
...
+```
+
+
+## JSON functions
+
+This section lists the built-in functions that operate on the JSON data type within Kùzu.
+
+### `to_json`
+
+Signature: `ANY -> JSON`
```
[the remark this answers is not in the document]
Comment (prrao87) [reply src/content/docs/extensions/json.mdx:547]: Yes, we should @acquamarin and @SterlingT3485

File: src/content/docs/extensions/json.mdx [hunk 6]
```diff
@@ -515,4 +515,315 @@ and `"to"` in the file, whose values point to the primary key values of the unde
...
+RETURN json_merge_patch('3', NULL);
+```
+Output:
+```json
+```
+(merge with a NULL path would result in NULL path)
+
+### `json_extract`
+
+Signatures: `JSON, STRING -> JSON`, `JSON, INTEGER -> JSON`
```
Comment (acquamarin) [inline src/content/docs/extensions/json.mdx:660]: I think we also support multipath json extract like:
```
json_extract(j, ['family', 'species']) 
```

File: src/content/docs/extensions/json.mdx [hunk 7]
```diff
@@ -515,4 +515,315 @@ and `"to"` in the file, whose values point to the primary key values of the unde
...
+Output:
+```json
+True
+```
+
+### `json_keys`
+
+Signature: `JSON -> STRING[]`
+
+`json_keys(json[])`
```
Comment (acquamarin) [inline src/content/docs/extensions/json.mdx:755]: I think it should be json_keys(json)`

File: src/content/docs/extensions/json.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (SterlingT3485) [reply src/content/docs/extensions/json.mdx:547]: let me write the doc for cast

File: src/content/docs/extensions/json.mdx (hunk 6 above)
Comment (SterlingT3485) [reply src/content/docs/extensions/json.mdx:660]: added the example query for multipath

File: src/content/docs/extensions/json.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (SterlingT3485) [reply src/content/docs/extensions/json.mdx:547]: added cast instruction

File: src/content/docs/extensions/json.mdx (hunk 7 above)
Comment (SterlingT3485) [reply src/content/docs/extensions/json.mdx:755]: changed

**Pull request merged**